src/panels/lovelace/hui-entity-filter-card.js

```diff
@@ -1,55 +1,114 @@
-import { html } from '@polymer/polymer/lib/utils/html-tag.js';
 import { PolymerElement } from '@polymer/polymer/polymer-element.js';
 
+import './hui-camera-preview-card.js';
 import './hui-entities-card.js';
+import './hui-entity-filter-card.js';
+import './hui-glance-card';
+import './hui-history-graph-card.js';
+import './hui-media-control-card.js';
+import './hui-picture-glance-card';
+import './hui-plant-status-card.js';
+import './hui-weather-forecast-card';
 
 import computeStateDomain from '../../common/entity/compute_state_domain.js';
+import computeCardElement from './common/compute-card-element.js';
 
 class HuiEntitiesCard extends PolymerElement {
-  static get template() {
-    return html`
-    <hui-entities-card
-      hass='[[hass]]'
-      config='[[_computeCardConfig(hass, config)]]'
-    ></hui-entities-card>
-`;
-  }
-
   static get properties() {
     return {
-      hass: Object,
-      config: Object,
+      hass: {
+        type: Object,
+        observer: '_hassChanged'
+      },
+      config: {
+        type: Object,
+        observer: '_configChanged'
+      }
     };
   }
 
   getCardSize() {
-    // +1 for the header
-    return 1 + this._getEntities(this.hass, this.config.filter).length;
+    return this.lastChild ? this.lastChild.getCardSize() : 1;
   }
 
-  // Return a list of entities based on a filter.
-  _getEntities(hass, filter) {
-    const filters = [];
+  // Return a list of entities based on filters.
+  _getEntities(hass, filterList) {
+    const entities = new Set();
+    filterList.forEach((filter) => {
+      const filters = [];
+      if (filter.domain) {
+        filters.push(stateObj => computeStateDomain(stateObj) === filter.domain);
+      }
+      if (filter.entity_id) {
+        filters.push(stateObj => this._filterEntityId(stateObj, filter.entity_id));
+      }
+      if (filter.state) {
+        filters.push(stateObj => stateObj.state === filter.state);
+      }
 
-    if (filter.domain) {
-      const domain = filter.domain;
-      filters.push(stateObj => computeStateDomain(stateObj) === domain);
+      Object.values(hass.states).forEach((stateObj) => {
+        if (filters.every(filterFunc => filterFunc(stateObj))) {
+          entities.add(stateObj.entity_id);
+        }
+      });
+    });
+    this._cardSize = entities.size;
```

Comment:
We should avoid functions that have side effects. `getCardSize` should just filter and return the length of the array.

Reply:
Wait, it's not even used ?

Reply:
Oh yes, I removed it